src/PlanktonIndividualExample.jl

```diff
@@ -0,0 +1,1280 @@
+### A Pluto.jl notebook ###
+# v0.15.0
+
+using Markdown
+using InteractiveUtils
+
+# ╔═╡ 153caed2-f735-4508-8e80-f0461a6af88c
+using PlanktonIndividuals, Plots
+
+# ╔═╡ 418c95a7-0125-40b6-acb5-e2011ea60df2
+md"""
+# [PlanktonIndividuals.jl](https://github.com/JuliaOcean/PlanktonIndividuals.jl) tutorial
+"""
+
+
+# ╔═╡ 2b46fec5-59bf-4d5b-8af6-0bd695ac1821
+md"""
+PlanktonIndividuals.jl is a fast individual-based model written in Julia that runs on both CPU and GPU. 
+
+It simulates the life cycle of ocean phytoplankton cells as Lagrangian particles while nutrients are represented as Eulerian tracers and advected over the gridded domain. 
+
+
+![phytoplankton](https://github.com/JuliaOcean/PlanktonIndividuals.jl/raw/master/docs/src/PI_Quota.jpeg)
+"""
+
+# ╔═╡ 6ec492fc-d622-11eb-2bfb-9d24a91685b7
+md"""
+Here we simulate phytoplankton cells as Lagrangian particles in a 2D flow field, with 
+one horizontal direction (x) and one vertical one (z), like in an ocean transect.
+
+The domain is periodic in the x direction while it is bounded in the z direction.
+"""
+
+# ╔═╡ 139108e4-0829-4710-8419-dc05808f173f
+md"""
+## 1. Import packages
+"""
+
+# ╔═╡ 15bc275c-7d15-479c-b644-d588e67cf3f5
+md"""
+Full documentation is available at [https://juliaocean.github.io/PlanktonIndividuals.jl/dev/](https://juliaocean.github.io/PlanktonIndividuals.jl/dev/)
+"""
+
+# ╔═╡ 3c05c373-af76-4d28-979c-e91aaa7321de
+p=dirname(pathof(PlanktonIndividuals));
+
+# ╔═╡ b7f4cd3e-78aa-44cd-bb6c-3f41ed9f8a96
+include(joinpath(p,"../examples/helper_functions.jl"));
+
+# ╔═╡ 37f42666-276f-43bf-b4b4-a245af7edb29
+md"""
+## 2. Architecture, Grid & Flow Fields
+"""
+
+# ╔═╡ 49c16379-d570-4849-bb74-0b7ccedb86a5
+md"""
+First we generate grid information (128 by 128 grid boxes, 1m thick, and 1m wide) and the computational architecture (CPU).
+
+[Speed-ups](https://juliaocean.github.io/PlanktonIndividuals.jl/dev/benchmarks/) on GPU.
+"""
+
+# ╔═╡ 6e45867e-e12e-4e44-9795-0a9ee01ecccb
+arch = CPU()
+
+# ╔═╡ 3b1861b8-4653-4c68-990d-7059dc187518
+grid = RegularRectilinearGrid(size=(128, 1, 128), spacing=(1, 1, 1))
+
+# ╔═╡ bee232fe-b10b-4ce8-81b6-6053611bd2e9
+md"""
+Then we use a stream function (see helper_functions.jl) to generate a simple flow field (displayed below) in a 2D vertical plane.
+"""
+
+# ╔═╡ bc16d909-f629-42b9-952b-777470b00a52
+(uvels, vvels, wvels, ϕcenters) = streamfunction_xz();
+
+# ╔═╡ 5f5a05c6-c76d-4f57-848b-73e391abe7dd
+md"""
+## 3. Model Setup
+"""
+
+# ╔═╡ 097232e8-dbe7-4c0e-8a3e-25638c28a623
+md"""
+Next we setup the individual-based model by specifying the architecture, grid, and plankton community.
+"""
+
+# ╔═╡ 467d1cfd-aadc-4e16-86d4-94cabedcd15a
+model = PlanktonModel(arch, grid; N_species = 1, 
```

Comment:
Thanks for pointing it out. `N_species = 3` doesn't work because no parameters are provided for the third species. In order to run a simulation with `N_species = 3` or more, you need to update the default parameters (see [here](https://juliaocean.github.io/PlanktonIndividuals.jl/dev/model_setup/#parameters)). An `n` elements `Array` is required for each parameter with `N_species = n`. `n = 2` by default, and that's why `N_species = 3` won't work.